client/packages/invoices/src/OutboundShipment/DetailView/OutboundLineEdit/allocation/utils.ts

```diff
@@ -0,0 +1,73 @@
+import { NumUtils } from '@common/utils';
+import { DraftStockOutLineFragment } from '../../../api/operations.generated';
+import { DateUtils } from '@common/intl';
+
+export const sumAvailableQuantity = (
+  draftLines: DraftStockOutLineFragment[]
+) => {
+  const sum = draftLines.reduce(
+    (acc, { stockLineOnHold, availablePacks, packSize, location }) =>
+      !location?.onHold && !stockLineOnHold
+        ? acc + availablePacks * packSize
+        : acc,
+    0
+  );
+
+  return sum;
+};
+
+export const getAllocatedUnits = ({
+  draftLines,
+  placeholderQuantity,
+}: {
+  draftLines: DraftStockOutLineFragment[];
+  placeholderQuantity: number | null;
+}) =>
+  NumUtils.round(
+    draftLines.reduce(
+      (acc, { numberOfPacks, packSize }) => acc + numberOfPacks * packSize,
+      0
+    ),
+    3
+  ) + (placeholderQuantity ?? 0);
+
+export const issueStock = (
+  draftLines: DraftStockOutLineFragment[],
+  idToIssue: string,
+  packs: number
+) => {
+  const foundRowIdx = draftLines.findIndex(({ id }) => id === idToIssue);
+  const foundRow = draftLines[foundRowIdx];
+  if (!foundRow) return draftLines;
+
+  const newDraftLines = [...draftLines];
+  newDraftLines[foundRowIdx] = {
```

Comment:
I feel like this is a mutation ? (since previous restructure doesn't copy draftLines, it just create a new array with them), I think for immutablility have to destructure down to primatives for objects that change, so somthing like [...sliceBefore, { ...found, newNumberOfPacks} ...sliceAfter]

Reply:
Ooh yeah great point, won't be entirely new objects, this was from existing logic... and it does look like everything is updating ok, maybe as we aren't doing any deeper comparisons anywhere?